### PR TITLE
Add support for container drivers, pod security, OpenShift

### PR DESCRIPTION
**Pull request by philgebhardt**
**Background**
[Gremlin 2.16.0][releasenotes] brought support for container runtimes: Cri-O and Containerd. It also brought support for the systemd cgroup driver as long as Gremlin mounts various hostPath resources (e.g. `/sys/fs/cgroup`, `/run/runc`)

**Change**
* Update Gremlin's Helm Chart to provision the Gremlin Daemonset according to the prerequisites of Gremlin 2.16.0.
* Maintain backward compatibility: introduce new helm values for opting in to new functionality; keep existing behavior unchanged
* Update READMEs, add examples

[releasenotes]: https://www.gremlin.com/docs/release-notes/overview/#2020-10-14

### CODE REVIEW

File: gremlin/README.md [hunk 1]
```diff
@@ -105,6 +124,7 @@ Install the Helm chart
 ```shell
 helm install gremlin/gremlin \
     --name gremlin \
+    --namespace gremlin \
```
Comment (philgebhardt) [inline gremlin/README.md:127]: these changes are not introduced as a result of chart changes, they were wrong on the current version of this README

**Review by Coding-Badly — APPROVED** [review]

**Review by gremsam — COMMENTED** [review]
This looks pretty good, I have some minor comments

File: gremlin/templates/daemonset.yaml [hunk 2]
```diff
@@ -116,11 +147,19 @@ spec:
           hostPath:
             path: /var/log/gremlin
         # If you want to run shutdown attacks on the host, the Gremlin Daemon requires a /proc/sysrq-trigger:/sysrq mount
-        - name: shutdown-trigger
-          hostPath:
-            path: /proc/sysrq-trigger
+        # - name: shutdown-trigger
```
[the remark this answers is not in the document]
Comment (philgebhardt) [reply gremlin/templates/daemonset.yaml:150]: This is a good catch. I think we do need to guard this with a flag. I will rework this part

File: gremlin/templates/gremlin-psp.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
+{{ if .Values.gremlin.podSecurity.podSecurityPolicy.create }}
+apiVersion: policy/v1beta1
```
[the remark this answers is not in the document]
Comment (philgebhardt) [reply gremlin/templates/gremlin-psp.yaml:2]: Yeah this is a good call out. I think I'd rather do this in a separate PR though unless this is brand new behavior, I'l have to look at and other `v1beta` apiVersions we might have.

File: gremlin/templates/daemonset.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (philgebhardt) [reply gremlin/templates/daemonset.yaml:150]: Thinking about this some more:

To preserve existing behavior, I'm going to return this back to previous behavior, adding this hard-coded mount back in, and adding it to the associated `gremlin-psp.yaml`. A followup iteration can make the act of mounting this path configurable. It was a mistake for me to combine this behavior change with the rest of this work.

**Pull request merged**